packages/js/src/ui/components/elements/Markdown.tsx

```diff
@@ -0,0 +1,71 @@
+import { createMemo, For, JSX } from 'solid-js';
+import { useStyle } from '../../helpers';
+
+interface Token {
+  type: 'bold' | 'text';
+  content: string;
+}
+
+const parseMarkdownIntoTokens = (text: string): Token[] => {
+  const tokens: Token[] = [];
+  let buffer = '';
+  let inBold = false;
+
+  for (let i = 0; i < text.length; i += 1) {
+    // Check if it's an escaped character
+    if (text[i] === '\\' && text[i + 1] === '*') {
+      buffer += '*';
+      i += 1;
+    }
+    // Check for bold marker **
+    else if (text[i] === '*' && text[i + 1] === '*') {
+      if (buffer) {
+        tokens.push({ type: inBold ? 'bold' : 'text', content: buffer });
+        buffer = '';
+      }
+      inBold = !inBold;
+      i += 1;
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Why manually incrementing the `i`?

‏

Reply:
ok nvm got it, you're also manually checking for `*` by using `i+1`